core/templates/dev/head/pages/exploration-editor-page/settings-tab/settings-tab.directive.ts

```diff
@@ -118,6 +118,324 @@ angular.module('oppia').directive('settingsTab', [
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance', function(
+                    $scope, $uibModalInstance) {
+                  $scope.transfer = $uibModalInstance.close;
+
+                  $scope.cancel = function() {
+                    $uibModalInstance.dismiss('cancel');
+                    AlertsService.clearWarnings();
+                  };
+                }
+              ]
+            }).result.then(function() {
+              ExplorationRightsService.makeCommunityOwned();
+            });
+          };
+
+          ctrl.deleteExploration = function() {
+            AlertsService.clearWarnings();
+
+            $uibModal.open({
+              templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                '/pages/exploration-editor-page/settings-tab/templates/' +
+                'delete-exploration-modal.template.html'),
+              backdrop: true,
+              controller: [
+                '$scope', '$uibModalInstance', function(
+                    $scope, $uibModalInstance) {
+                  $scope.reallyDelete = $uibModalInstance.close;
+
+                  $scope.cancel = function() {
+                    $uibModalInstance.dismiss('cancel');
+                    AlertsService.clearWarnings();
+                  };
+                }
+              ]
+            }).result.then(function() {
+              EditableExplorationBackendApiService.deleteExploration(
+                ctrl.explorationId).then(function() {
+                $window.location = CREATOR_DASHBOARD_PAGE_URL;
+              });
+            }, function() {
+              // Promise is returned by uimodal which is handled here.
+              // No further action is needed.
+            });
+          };
+
+          ctrl.unpublishExplorationAsModerator = function() {
+            AlertsService.clearWarnings();
+
+            var moderatorEmailDraftUrl = '/moderatorhandler/email_draft';
+
+            $http.get(moderatorEmailDraftUrl).then(function(response) {
+              // If the draft email body is empty, email functionality will
+              // not be exposed to the mdoerator.
+              var draftEmailBody = response.data.draft_email_body;
+
+              $uibModal.open({
+                bindToController: {},
+                templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+                  '/pages/exploration-editor-page/settings-tab/templates/' +
+                  'moderator-unpublish-exploration-modal.template.html'),
+                backdrop: true,
+                resolve: {
+                  draftEmailBody: function() {
+                    return draftEmailBody;
+                  }
+                },
+                controllerAs: '$ctrl',
+                controller: [
+                  '$uibModalInstance', 'draftEmailBody',
+                  function($uibModalInstance, draftEmailBody) {
+                    var ctrl = this;
```

Comment:
For these inner things, should we still be using ctrl, or would modalCtrl be better? Don't want to confuse it with the one outside.